### PR TITLE
move most volume panel functionality into css state

### PR DESCRIPTION
**Pull request by gkatsev**
## Description
Built from https://github.com/videojs/video.js/pull/3958
Moves as much transition and "logic" into CSS as possible.
Use some javascript to make sure that things are sized properly when tabbing with the keyboard.

### CODE REVIEW

**Comment by BrandonOCasey** [comment]
I am not seeing the `VolumeControl`/`VolumeBar` being hidden on Firefox/Chorme

**Comment by gkatsev** [comment]
As in if you're just adding the volume control directly? Those are always shown unless when part of the volume panel.

**Comment by BrandonOCasey** [comment]
seems like it was a cache issue, I see it working now

File: src/js/control-bar/volume-panel.js [hunk 1]
```diff
@@ -45,16 +45,20 @@ class VolumePanel extends Component {
...
-    // show/hide the VolumeControl (slider/bar)
-    this.children().forEach((child) => {
-      this.on(child, ['focus'], () => this.volumeControl.show());
-      this.on(child, ['blur'], () => this.volumeControl.hide());
-    });
+    this.on(this.volumeControl, ['sliderinactive'], this.sliderInactive_);
+    this.on(this.muteToggle, 'blur', this.sliderInactive_);
+  }
+
+  sliderActive_() {
```
Comment (BrandonOCasey) [inline src/js/control-bar/volume-panel.js]: These will need jsdoc comments or maybe these should just be local functions?

**Comment by BrandonOCasey** [comment]
I am not seeing the horizontal `VolumeControl`/`VolumeBar` hide on `IE9 Windows 7`(HTML) or `IE8 Windows 7` (Flash)

**Comment by gkatsev** [comment]
Horizontal volume control isn't hidden on IE8 and 9. Always expanded.

**Comment by BrandonOCasey** [comment]
ah ok well then other than my comment it LGTM

**Comment by BrandonOCasey** [comment]
Actually one last thing, I am not seeing the vertical volume bar hide on IE8. Works on IE9

**Comment by BrandonOCasey** [comment]
@gkatsev tested, the only issue as you said is the missing ball on IE8. I think thats acceptable.

**Pull request merged**